### ISSUE TITLE
where should the training files be placed?

### ISSUE THREAD

**Issue by xia00100** [LoSealL/VideoSuperResolution#5]
I'm really appreciate your sharing of this project . But I just started using Tensorflow, so May i ask where should the training files be placed？ and I already download 91-Image/SET 5/SET 14 etc. but I can't run it success!

**Comment by xia00100** [comment]
InvalidArgumentError (see above for traceback): You must feed a value for placeholder tensor 'label/hr' with dtype float and shape [?,?,?,1]
         [[node label/hr (defined at /mnt/md0/xianan/VideoSuperResolution-master/Train/VSR/Framework/SuperResolution.py:124)  = Placeholder[dtype=DT_FLOAT, shape=[?,?,?,1], _device="/job:localhost/replica:0/task:0/device:GPU:0"]()]]
         [[{{node loss/Const_2/_39}} = _Recv[client_terminated=false, recv_device="/job:localhost/replica:0/task:0/device:CPU:0", send_device="/job:localhost/replica:0/task:0/device:GPU:0", send_device_incarnation=1, tensor_name="edge_279_loss/Const_2", tensor_type=DT_INT32, _device="/job:localhost/replica:0/task:0/device:CPU:0"]()]]

**Comment by LoSealL** [comment]
@xia00100 Hi, you can check [datasets.yaml](https://github.com/LoSealL/VideoSuperResolution/blob/master/Data/datasets.yaml).

let's say you download **91-image.tar.gz** and decompress 91 images into `/mnt/data/datasets/91-image`, you just modify `root` and `Path: 91-image` sections in [datasets.yaml](https://github.com/LoSealL/VideoSuperResolution/blob/master/Data/datasets.yaml). If you are familiar with `Path.glob()` function, it should be easy to understand.
After 'Root' and 'Path' are correctly set, combined them in `Dataset` section as I did.

At last, VSR can automatically parse datasets' files and start training.

**Comment by LoSealL** [comment]
Currently, try training ESPCN for a good to start.
If you want to train SRCNN and VDSR, currently add `--add_custom_callbacks=upsample` as an argument for `run.py`.

**Comment by LoSealL** [comment]
@Iatboo Did you use your VESPCN or my version? In paper, the author claimed optical flow ranges (-1, 1), representing movement across the entire image, which however leads to quite poor results.
So instead I inteprete optical flow ranges (-1, 1) as the absolute pixel, which means the movement is restricted up to 1 pixel.
By doing that way, my result is around 25.5dB (VID4), better than **bicubic**, **EDSR**, worse than **Bayesian**.

I trained with MCL-V and GOPRO dataset, you can find the download link in README file.

*The pre-trained weights is on-the-way, please wait for some days (or some months...)*

**Comment by Iatboo** [comment]
@LoSealL Thank you very much for your detailed answer.

**Comment by LoSealL** [comment]
closed since no more questions here
